port all JIT tests to gtest

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45264 port all JIT tests to gtest**

Context for why we are porting to gtest in: https://github.com/pytorch/pytorch/pull/45018.

This PR completes the process of porting and removes unused files/macros.

Differential Revision: [D23901392](https://our.internmc.facebook.com/intern/diff/D23901392)

suo committed:
port all JIT tests to gtest

[ghstack-poisoned]

suo committed:
Update on "port all JIT tests to gtest"


Differential Revision: [D23901392](https://our.internmc.facebook.com/intern/diff/D23901392)

[ghstack-poisoned]

suo commented:
cc @csarofeen: heads up that this PR changes the GPU fuser tests to be registered through gtest only. So running cpp tests through the `test_jit.py` will no longer work. Let me know that is an issue!

suo committed:
Update on "port all JIT tests to gtest"


Context for why we are porting to gtest in: https://github.com/pytorch/pytorch/pull/45018.

This PR completes the process of porting and removes unused files/macros.

Differential Revision: [D23901392](https://our.internmc.facebook.com/intern/diff/D23901392)

[ghstack-poisoned]

suo committed:
Update on "port all JIT tests to gtest"


Context for why we are porting to gtest in: https://github.com/pytorch/pytorch/pull/45018.

This PR completes the process of porting and removes unused files/macros.

Differential Revision: [D23901392](https://our.internmc.facebook.com/intern/diff/D23901392)

[ghstack-poisoned]

jjsjann123 commented:
> So running cpp tests through the test_jit.py will no longer work. Let me know that is an issue!

IIUC, CI would still run cpp tests, but jut not via the python script?
I don't fully understand how our own CI works, but I suspect it's relying on `test_jit.py` to cover our cpp tests. Can you point me to where cpp tests are run in upstream CI? so I can mimic that in our own :)

suo commented:
> IIUC, CI would still run cpp tests, but jut not via the python script?
Correct, our upstream CI runs the JIT tests here: https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L200. Adding a similar line in your CI should be sufficient.

ZolotukhinM commented on aten/src/ATen/test/thread_init_test.cpp:
```diff
@@ -1,6 +1,6 @@
 #include <ATen/ATen.h>
 #include <ATen/Parallel.h>
-#include <test/cpp/jit/test_base.h>
+#include <test/cpp/tensorexpr/test_base.h>
```
This looks kinda awkward. Why does this depend on something in tensorexpr now? :)

suo replied:
This test depends on some of the gtest polyfills in `test_base.h`. Since I am deleting the JIT's version, I included the `tensorexpr` copy to avoid editing this test too much. I think it wouldn't be too hard to either 1) include gtest or 2) just define the polyfills here, but I didn't want to make this PR bigger in non-JIT places.

ZolotukhinM commented on test/cpp/jit/test_interpreter.cpp:
```diff
@@ -20,88 +26,97 @@ graph(%a.1 : Tensor,
+  ASSERT_FALSE(stack[2].toBool());
+}
+
+// TODO: These tests weren't doing anything.
```
Do these tests pass if we enable them now?

suo replied:
There were a few tests that weren't running, because they were defined in a test file but not registered in `tests.h`. If they are commented out, it's because they failed when I re-enabled them :(

ZolotukhinM reviewed (approved):
Mostly looks good, my only question is about `thread_init_test.cpp` (see below).

suo committed:
Update on "port all JIT tests to gtest"


Context for why we are porting to gtest in: https://github.com/pytorch/pytorch/pull/45018.

This PR completes the process of porting and removes unused files/macros.

Differential Revision: [D23901392](https://our.internmc.facebook.com/intern/diff/D23901392)

[ghstack-poisoned]

csarofeen commented:
@suo  you mentioned a conflict, can we give you a hand? CC @jjsjann123

suo commented:
> @suo you mentioned a conflict, can we give you a hand? CC @jjsjann123

no worries, I just rebased and fixed the conflicts manually. Thanks!
